nntrainer/layers/pooling2d_layer.cpp

```diff
@@ -125,6 +143,13 @@ void Pooling2DLayer::forwarding(RunLayerContext &context, bool training) {
 }
 
 void Pooling2DLayer::calcDerivative(RunLayerContext &context) {
+  std::vector<props::PoolSize> pool_size =
```

Comment:
How about using `std::vector<props::PoolSize>& pool_size` to avoid copy?

Reply:
Done